docs/_docs/reference/changed-features/pattern-matching.md

```diff
@@ -162,27 +142,66 @@ object Nat:
 // 5 is a natural number
 ```
 
-## Name-based Match
+### Name-based Match
 
-- `N > 1` is the maximum number of consecutive (parameterless `def` or `val`) `_1: P1 ... _N: PN` members in `U`
+- `S` has `N > 1` members such that they are each `val`s or parameterless `def`s, and named from `_1` with type `P1` to `_N` with type `PN`
+- `S` doesn't have `N+1` members satisfying the previous point, `N` is maximal
 - Pattern-matching on exactly `N` patterns with types `P1, P2, ..., PN`
 
+For example, where `AlwaysEmpty <: R`, `S` is `NameBased`:
 ```scala
-object ProdEmpty:
+object MyPatternMatcher:
+  def unapply(s: String) = AlwaysEmpty
+
+object AlwaysEmpty:
+  def isEmpty = true
+  def get = NameBased
+
+object NameBased:
   def _1: Int = ???
   def _2: String = ???
-  def isEmpty = true
-  def unapply(s: String): this.type = this
-  def get = this
 
 "" match
-  case ProdEmpty(_, _) => ???
+  case MyPatternMatcher(_, _) => ???
   case _ => ()
 ```
 
-## Sequence Match
+## Variadic Extractors
+
+Variadic extractors expose the following signature:
+
+```scala
+def unapplySeq[A](x: T)(implicit x: B): U
+```
+
+The type `U` conforms to one of the following matches:
+
+- sequence match
+- product-sequence match
+
+Or `U` conforms to the type `R`:
+
+```scala
+type R = {
+  def isEmpty: Boolean
+  def get: S
+}
+```
+
+and `S` conforms to one of the two matches above.
+
+The former form of `unapplySeq` has higher priority, and _sequence match_ has higher
+precedence over _product-sequence match_.
 
-- `U <: X`, `T2` and `T3` conform to `T1`
+A usage of a variadic extractor is irrefutable if one of the following conditions holds:
+
+- the extractor is used directly as a sequence match or product-sequence match
+- `U = Some[T]` (for Scala 2 compatibility)
+- `U <: R` and `U <: { def isEmpty: false }`
+
+### Sequence Match
+
+- `U <: X` (resp `S <: X`)
```

Comment:
"resp" to what?

Reply:
resp to
> The type `U` conforms to one of the following matches:
>
> - sequence match
> - product-sequence match
>
> Or `U` conforms to the type `R`: [...] `S` [..]

The fact the sequence match is used both for `U` as a whole, or a part of it `S` makes it very awkward to write the rules, I used "resp" to highlight this double nature, but am taking any advice

Reply:
@sjrd Given the above, does it make more sense/do you have any ideas on how we could improve it ?